pkg/utils/node/predicates.go

```diff
@@ -42,13 +47,27 @@ func IsPastTTL(node *v1.Node) bool {
 	return time.Now().After(ttlTime)
 }
 
-// TODO: implement underutilized function (some generalized predicate)
-func IsUnderutilized(pods []*v1.Pod) bool {
-	counter := 0
+// IsUnderutilized returns true if all of a node's pod's resource requests sum to less than the given thresholds
+func IsUnderutilized(pods []*v1.Pod, threshold map[v1.ResourceName]float64) bool {
```

Comment:
Consider using a Quantity instead of a float64.

Reply:
or a ResourceList